src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -184,26 +184,27 @@ trait Qualif {
         cx: &ConstCx<'_, 'tcx>,
         place: PlaceRef<'_, 'tcx>,
     ) -> bool {
-        let proj = place.projection.as_ref().unwrap();
+        let elem = &place.projection[place.projection.len() - 1];
```

Comment:
Invent a function to split a slice into `[init @ .., last]`? (this idea is repeated a few times in this PR...)